### PR TITLE
Week8

### PR DESCRIPTION
**Pull request by shbh29**
PTAL

Added:
 - Linked List Base implementation
 - Stack Base Implementation
 - Next Greater Number Problem solved using stack
 - Queue implemented using 2 stack
 - Reverse Linked List implemented.
 - Rotate Linked List by Given number problem
 - Parenthesis Matching Problem

### CODE REVIEW

File: Week-7/Ex1.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+let ll = new LLReverse();
+ll.add(1);
+ll.add(2);
+ll.add(3);
+ll.add(4);
+ll.print();
+ll.reverseList();
```
Comment (imranmd) [inline Week-7/Ex1.js:34]: Good

File: Week-7/Ex2.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Rotate By given number
+// SC : O(1)
+// TC: O(n)
+const LinkedList = require('./LinkedListSkeleton');
```
Comment (imranmd) [inline Week-7/Ex2.js:4]: Good, Like the way of reusing

File: Week-7/Ex3.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+let ll = new LLCircular();
+ll.add(1);
+ll.add(2);
+ll.add(3);
+ll.add(4);
+
+ll.print();
+ll.pointLastNodeTo(pointLastNodeTo); // point to node with data 2 and index 1;
+
+console.log(`Is this List Circular? : ${ll.isCircular()}`);
```
Comment (imranmd) [inline Week-7/Ex3.js:43]: Good

File: Week-7/Ex4.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    return stack.isEmpty();
+}
+
+
+// let expression = "{[]}()({[]})";
+
+let expression = "{[]}()"
+// let expression = "{[]}()({[}])"
+console.log(`paranthesis matching result for expression
```
Comment (imranmd) [inline Week-7/Ex4.js:53]: Good

File: package.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    "cjs-module-lexer": "^1.2.2",
+    "collect-v8-coverage": "^1.0.1",
+    "color-convert": "^2.0.1",
+    "color-name": "^1.1.4",
+    "concat-map": "^0.0.1",
+    "convert-source-map": "^1.9.0",
+    "cross-spawn": "^7.0.3",
+    "debug": "^4.3.4",
+    "dedent": "^0.7.0",
+    "deepmerge": "^4.2.2",
```
Comment (imranmd) [inline package.json:40]: Why all these dependencies are required

File: Week-7/Ex6.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        this.popStack = new Stack();
+    }
+    push(no) {
+        this.pushStack.push(no);
+        console.log(`pushed: ${no}`);
+    }
+    pop() {
+        if(this.popStack.isEmpty()) {
+            while(!this.pushStack.isEmpty()) {
+                this.popStack.push(this.pushStack.pop());
```
Comment (imranmd) [inline Week-7/Ex6.js:19]: Good

**Review by imranmd — APPROVED** [review]
Overall Awesome work @shbh29

**Comment by shbh29** [comment]
Thanks @imranmd

**Pull request merged**